app/models/subscriber.rb

```diff
@@ -0,0 +1,32 @@
+class Subscriber
+  def self.for_discussion(discussion)
+    new(discussion)
+  end
+
+  def initialize(discussion)
+    @discussion = discussion
+    @subscriptions = Subscription.where(discussion_id: discussion.id)
+  end
+
+  def subscribe(users)
+    users.each do |user|
+      create_subscription_for_user(user)
+    end
+  end
+
+  private
+
+  attr_reader :discussion, :subscriptions
+
+  def create_subscription_for_user(user)
+    return if user_already_subscribed?(user)
+
+    Subscription.create(user: user, discussion: discussion)
+  end
+
+  def user_already_subscribed?(user)
+    subscriptions.any? do |subscription|
+      subscription.user_id == user.id
+    end
+  end
```

Comment:
I'm not saying that it is not possible with AR, I assumed it was. But does it really optimize anything in terms of DB calls, that was what I was asking. But I can test that I let you know.